Tools/Scripts/libraries/resultsdbpy/resultsdbpy/view/static/js/timeline.js

```diff
@@ -512,6 +555,111 @@ class TimelineFromEndpoint {
+                else
+                    lastSuccessDot = null;
+                failedScopes.set(config, {
+                    initFailureResult: dots[0],
+                    initFailure: {initFailureType, initFailureNumber},
+                    lastSuccessDot
+                });
+            });
+
+            if (!failedScopes.size) {
+                alert('No failures detect in selected results');
+                return;
+            }
+
+            let titleComponentSet = new Set();
+            let descriptionComponent = [];
+            Array.from(failedScopes.keys()).forEach(config => {
+                const failedScope = failedScopes.get(config);
+                const versionName = config.version_name ? config.version_name : Configuration.integerToVersion(config.version);
+                titleComponentSet.add(`${failedScope.initFailure.initFailureType} on ${versionName}(${config.model})`);
+                descriptionComponent.push(`${failedScope.initFailure.initFailureNumber ? `${failedScope.initFailure.initFailureNumber} ${failedScope.initFailure.initFailureType}` : ''}`);
+                descriptionComponent.push(`Hardware:     \t${config.model}`);
+                descriptionComponent.push(`Architecture: \t${config.architecture}`);
+                descriptionComponent.push(`OS:           \t${versionName}`);
+                descriptionComponent.push(`Style:        \t${config.style}`);
+                if (config.suite === 'layout-tests')
+                    descriptionComponent.push(`Flavor        \t${config.flavor}`);
+                if (config.sdk)
+                    descriptionComponent.push(`SDK:          \t${config.sdk}`);
+                descriptionComponent.push('---------------------');
+                descriptionComponent.push(`Most recent failures: ${CommitBank.commitsDuring(failedScope.initFailureResult.uuid).map(commit => commit.label()).join(' ,')}`);
+                descriptionComponent.push(this.getTestResultUrl(config, failedScope.initFailureResult));
+                descriptionComponent.push('---------------------');
+                if (failedScope.lastSuccessDot) {
+                    descriptionComponent.push(`Last success: ${CommitBank.commitsDuring(failedScope.lastSuccessDot.uuid).map(commit => commit.label()).join(' ,')}`);
+                    descriptionComponent.push(this.getTestResultUrl(config, failedScope.lastSuccessDot));
+                }
+            });
+
+            let component = 'WebKit';
+            let version = 'All';
+            switch (this.suite) {
+                case 'webkitpy-tests':
+                    component = 'WebKit Tools';
+                break;
+                case 'internal-media-tests':
+                    component = 'WebKit Media';
+                break;//
+                case 'javascriptcore-tests':
+                    component = 'JavaScriptCore';
+                break;
+            }
+            const radarUrlComponents = {
```

Comment:
I think we need to create a more generic API for this.

OpenSource clearly needs to generate bugzilla links instead of radars. Also, things like component setting belong in the application layer, not the library, which will complicate things.

First step for generalizing this is probably to create a generic "createBug" function template taking `title`, `description`, and `classification` for sure, maybe `component` too? You will also probably need to provide some kind of filtering callback so the application instantiating the library can specify it's own components.

Reply:
This is application layer already. this is not part of library

Reply:
I guess we actually have three layers in this application:
- JavaScript libraries
- resultsdbpy "library" that is a packaged application configured by the caller
- Cloud hosted specific application
I'm saying that this code setting components based on test suites belong a level up in the "cloud-hosted specific application." Otherwise this would be the first part of resultsdbpy that references a specific suite.